src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      for (const o in state) {
+        delete state[o];
+      };
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const i of action.keysToRemove) {
+        delete state[i];
+      };
+    }
+
+    if (action.type === 'addProperties') {
+      for (const [key, value] of Object.entries(action.extraData)) {
+        state[key] = value;
+      }
+    }
+  }
```

Comment:
1. use switch...sace instead of if blocks
2. o and i - bad naming, use more meanfull names